tidy3d/components/medium.py

```diff
@@ -967,14 +1459,105 @@ def from_dispersion(cls, n: float, freq: float, dn_dwvl: float = 0, **kwargs):
-        b_coeff = (wvl**2 - c_coeff) / wvl**2 * nsqm1
-        coeffs = [(b_coeff, c_coeff)]
 
-        return cls(coeffs=coeffs, **kwargs)
+class CustomSellmeier(CustomDispersiveMedium, Sellmeier):
+    """A spatially varying dispersive medium described by the Sellmeier model.
+    The frequency-dependence of the refractive index is described by:
+
+    Note
+    ----
+    .. math::
+
+        n(\\lambda)^2 = 1 + \\sum_i \\frac{B_i \\lambda^2}{\\lambda^2 - C_i}
+
+    Example
+    -------
+    >>> x = np.linspace(-1, 1, 5)
+    >>> y = np.linspace(-1, 1, 6)
+    >>> z = np.linspace(-1, 1, 7)
+    >>> b1 = SpatialDataArray(np.random.random((5, 6, 7)), coords=dict(x=x, y=y, z=z))
+    >>> c1 = SpatialDataArray(np.random.random((5, 6, 7)), coords=dict(x=x, y=y, z=z))
+    >>> sellmeier_medium = CustomSellmeier(coeffs=[(b1,c1),])
+    >>> eps = sellmeier_medium.eps_model(200e12)
+    """
+
+    coeffs: Tuple[Tuple[SpatialDataArray, SpatialDataArray], ...] = pd.Field(
+        title="Coefficients",
+        description="List of Sellmeier (:math:`B_i, C_i`) coefficients.",
+        units=(None, MICROMETER + "^2"),
+    )
+
+    @pd.validator("coeffs", always=True)
+    def _correct_shape_and_sign(cls, val):
+        """every term in coeffs should have the same shape, and B>=0 and C>0."""
+        expected_shape = val[0][0].shape
+        for (B, C) in val:
+            if B.shape != expected_shape or C.shape != expected_shape:
+                raise SetupError("Every term in 'coeffs' should have the same dimension.")
+            if not CustomDispersiveMedium._validate_isreal_dataarray_tuple((B, C)):
+                raise SetupError("'B' and 'C' should be real.")
+            if np.any(B < 0):
+                raise SetupError("'B' should be non-negative.")
+            if np.any(C <= 0):
+                raise SetupError("'C' should be positive.")
+        return val
+
+    def _pole_residue_dict(self) -> Dict:
+        """Dict representation of Medium as a pole-residue model."""
+        poles_dict = Sellmeier._pole_residue_dict(self)
+        poles_dict.update({"eps_inf": xr.ones_like(self.coeffs[0][0])})
+        return poles_dict
+
+    def eps_dataarray_freq(
+        self, frequency: float
+    ) -> Tuple[SpatialDataArray, SpatialDataArray, SpatialDataArray]:
+        """Permittivity array at ``frequency``.
+
+        Parameters
+        ----------
+        frequency : float
+            Frequency to evaluate permittivity at (Hz).
+
+        Returns
+        -------
+        Tuple[:class:`.SpatialDataArray`, :class:`.SpatialDataArray`, :class:`.SpatialDataArray`]
+            The permittivity evaluated at ``frequency``.
+        """
+        eps = Sellmeier.eps_model(self, frequency)
+        return (eps, eps, eps)
+
+    @classmethod
+    def from_dispersion(
+        cls, n: SpatialDataArray, freq: float, dn_dwvl: SpatialDataArray, **kwargs
+    ):  # pylint:disable=signature-differs
```

Comment:
The signature of this function in `CustomSellmeier` is from the one in `Sellmeier`: "n" and "dn_dwvl" are now SpatialDataArray. So pylint will complain. Do you think it's fine? @tylerflex

Reply:
I think it's ok.